MaterialLib/SolidModels/Ehlers-impl.h

```diff
@@ -668,22 +675,28 @@ bool SolidEhlers<DisplacementDim>::computeConstitutiveRelation(
             .noalias() = calculateDResidualDEps<DisplacementDim>(K, G);
 
         if (_damage_properties)
-            updateDamage(t, x, _state);
+            updateDamage(t, x, state);
 
         // Extract consistent tangent.
-        C.noalias() =
+        *tangentStiffness =
             _mp.G(t, x)[0] *
             linear_solver.solve(-dresidual_deps)
                 .template block<KelvinVectorSize, KelvinVectorSize>(0, 0);
     }
 
-    // Update sigma.
     if (_damage_properties)
-        sigma_final = G * sigma * (1 - _state.damage.damage);
-    else
-        sigma_final.noalias() = G * sigma;
-
-    return true;
+        return {_mp.G(t, x)[0] * sigma * (1 - state.damage.damage),
+                std::unique_ptr<typename MechanicsBase<
+                    DisplacementDim>::MaterialStateVariables>{
+                    new MaterialStateVariables{state}},
+                std::move(tangentStiffness)};
+
+    return {
+        _mp.G(t, x)[0] * sigma,
```

Comment:
Only first component of the tuple is different from the case above. So some lines of code could be saved.